Address review comments from Mark Nottingham (#334)
MikeBishop/dns-alt-svc: draft-ietf-dnsop-svcb-https.md

The "alpn" and "no-default-alpn" SvcParamKeys together
indicate the set of Application Layer Protocol Negotiation (ALPN)
protocol identifiers {{!ALPN=RFC7301}}
and associated transport protocols supported by this service endpoint (the
"SVCB ALPN set").

As with Alt-Svc {{AltSvc}}, each ALPN protocol identifier is used to
identify the application protocol and associated suite
of protocols supported by the endpoint (the "protocol suite").

> The presence of an ALPN protocol in the SVCB ALPN set indicates that this

Reviewer: ```suggestion
The presence of an ALPN protocol identifier in the SVCB ALPN set indicates that this
```

service endpoint, described by TargetName and the other parameters (e.g.

> "port") offers service with the protocol suite associated with this ALPN
> protocol.

Reviewer: ```suggestion
"port") offers service with the protocol suite associated with this ALPN identifier.
```
In both cases, there are no ALPN protocols, just identifiers that reference protocols.

Clients filter the set of ALPN identifiers to match the protocol suites they
support, and this informs the underlying transport protocol used (such

> as QUIC-over-UDP or TLS-over-TCP).  ALPN protocol names that do not uniquely

Reviewer: ```suggestion
as QUIC-over-UDP or TLS-over-TCP).  ALPN protocol identifiers that do not uniquely
```

## HTTP Strict Transport Security {#hsts}

By publishing a usable HTTPS RR, the server operator indicates that all

> useful HTTP resources on this Host ({{!I-D.draft-ietf-httpbis-semantics,
> Section 7.2}}) are reachable over HTTPS, similar to HTTP Strict Transport

Reviewer: ```suggestion
origins with an "http" scheme governed by this host ({{!I-D.draft-ietf-httpbis-semantics,
Section 4.3.2}}) have an equivalent origin with an "https" scheme ({{!I-D.draft-ietf-httpbis-semantics,
Section 4.3.3}}), similar to HTTP Strict Transport
```

I think Mark's point is that we're making a statement about mapping between one set of origins and another, not individual resources or a single origin.  The origin transformation rules imported from HSTS below reflect this.

Author: I'm sure your terminology is an improvement, but I think we need to tweak this.

My concern with this phrasing is that "equivalent origin" suggests that the resources on both origins should be the same, but that's not actually what we want.  Ideally, the HTTP origin will be blank except for a redirect to HTTPS.  That's the rationale for the current language about "all useful HTTP resources".

Can you think of a way to represent this idea without implying the that origins have the same content?

Reviewer 2: "By publishing a usable HTTPS RR, the server operator indicates that all origins with an "http" scheme governed by this host have HTTPS resources have an origin with the correct "https" scheme similar to HTTP Strict Transport"

no that sucks too.   I find "equivalent" an issue as well after reading it over a few trimes.

Reviewer: Seems like what we want is to communicate that you're going to end up with the same content either way.  That might be because the http:// endpoint serves content, or it might be because it sends you a 3XX telling you that you should have done https://.

If the issue is with the term "equivalent," how about "associated"?  The origins are cooperating, they will get you to the same content (though the mechanism may vary), but they may not return identical responses.

Reviewer 2: I like associated.

Author: I've tried to rewrite this, borrowing more text from the HSTS RFC and some of @MikeBishop's language.  It now reads

>An HTTPS RR directs the client to communicate with this host only over a
> secure transport, similar to HTTP Strict Transport Security {{HSTS}}.

...

> If the "http" origin exists and contains important resources that have no
> equivalent at the "https" origin, the operator MUST NOT publish an HTTPS RR.

This construction is equivalent to {{Section 8.3 of HSTS}}, point 5.

> If the "http" origin exists and contains important resources that have no

Reviewer: I feel like "important" may be a sticking point here.  I infer what you mean, but there's no definition of "important" that someone could point to confirm that they are compliant with this MUST.  Is it sufficient to simply say "contains resources"?  The content of a 301 isn't a resource, it's merely the payload of a redirect.

Author: Here's another variation:

>If this redirection would result in a loss of functionality (e.g. important
resources that are only available on the "http" origin), the operator MUST
NOT publish an HTTPS RR.
